Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -1649,18 +1660,114 @@ - (void)handleInitSuccess {
             branchSharingController.deepLinkingCompletionDelegate = self;
             self.deepLinkPresentingController = [[[UIApplicationClass sharedApplication].delegate window] rootViewController];
             
-            if ([self.deepLinkPresentingController presentedViewController]) {
-                [self.deepLinkPresentingController dismissViewControllerAnimated:NO completion:^{
-                    [self.deepLinkPresentingController presentViewController:branchSharingController animated:YES completion:NULL];
-                }];
+            if([self.deepLinkControllers[key] isKindOfClass:[BranchDeepLinkModel class]]) {
+                BranchDeepLinkModel* deepLinkModel = self.deepLinkControllers[key];
+                UIViewController <BranchDeepLinkingController> *branchSharingController = deepLinkModel.viewController;
+                
+                if ([branchSharingController respondsToSelector:@selector(configureControlWithData:)]) {
+                    [branchSharingController configureControlWithData:latestReferringParams];
+                }
+                else {
+                    BNCLogWarning(@"[Branch Warning] View controller does not implement configureControlWithData:");
```

Comment:
The `BNCLog` functions already format `[Branch Warning]` in a uniform way so they aren't needed in the message.